mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerImpl.java

```diff
@@ -135,23 +134,27 @@ private void handleFullReindexingIterations(List<MailboxPath> mailboxPaths, Glob
     private void handleMailboxIndexingIterations(MailboxSession mailboxSession, MailboxRegistration mailboxRegistration, Mailbox mailbox, Iterator<MailboxMessage> iterator) throws MailboxException {
         while (iterator.hasNext()) {
             MailboxMessage message = iterator.next();
-            ImpactingMessageEvent impactingMessageEvent = findMostRelevant(mailboxRegistration.getImpactingEvents(message.getUid()));
+            Optional<ImpactingMessageEvent> impactingMessageEvent = findMostRelevant(mailboxRegistration.getImpactingEvents(message.getUid()));
 
-            impactingMessageEvent.newFlags().ifPresent(message::setFlags);
+            impactingMessageEvent.flatMap(ImpactingMessageEvent::newFlags).ifPresent(message::setFlags);
 
-            if (!impactingMessageEvent.wasDeleted()) {
+            if (wasNotDeleted(impactingMessageEvent)) {
```

Comment:
please keep positive method naming  `wasDeleted`, it makes the code more easy to understand (wasNotDeleted return true is always a challenge to track for my mind)